### PR TITLE
feat(proto)!: Condense redundant queries into market_summary

### PR DESCRIPTION
**Pull request by toteki**
## Description

- Removes individual queries which are already fields in `market_summary`.
- Adds new fields to `market_summary`.
- Updates tests.
- Leaves TODOs for implementations of the token market limits #1162 and #1163

closes: #1164 

---

### Author Checklist

_All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues._

I have...

- [x] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [x] added `!` to the type prefix if API or client breaking change
- [x] added appropriate labels to the PR
- [x] targeted the correct branch (see [PR Targeting](https://github.com/umee-network/umee/blob/main/CONTRIBUTING.md#pr-targeting))
- [x] provided a link to the relevant issue or specification
- [x] added a changelog entry to `CHANGELOG.md`
- [x] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [x] updated the relevant documentation or specification
- [x] reviewed "Files changed" and left comments if necessary
- [x] confirmed all CI checks have passed

### Reviewers Checklist

_All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items._

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed all author checklist items have been addressed
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

File: x/leverage/spec/03_queries.md [hunk 1]
```diff
@@ -8,15 +8,7 @@ General queries:
...
-- **Borrow APY** queries for the [Borrow APY](01_concepts.md#Borrow-APY) of a specified denomination.
-- **Supply APY** queries for the [Supplying APY](01_concepts.md#Supplying-APY) of a specified denomination.
-- **Reserve Amount** queries for the amount reserved of a specified denomination.
-- **Total Borrowed** queries for the total borrowed amount of a specified token denomination.
-- **Total Collateral** queries for the total collateral amount of a specified uToken denomination.
-- **Exchange Rate** queries the [uToken Exchange Rate](01_concepts.md#uToken-Exchange-Rate) of a given uToken denomination.
-- **Total Supplied** queries the [Total Supplied](01_concepts.md#Total-Supplied) of a specified denomination.
-- **Total Supplied Value** queries the equivalent USD value of [Total Supplied](01_concepts.md#Total-Supplied) of a specified denomination.
-- **Market Summary** combines several asset-specifying queries for more efficient frontend access.
+- **Market Summary** collects data on a given `Token` denomination. A description of each response field can be found in the [QueryMarketSummaryResponse proto definition](../../../proto/umee/leverage/v1/query.proto)
```
Comment (toteki) [inline x/leverage/spec/03_queries.md:11]: Since the line numbers in the destination file are unstable whether I use the `proto` or the `pb.go`, the best I can do right now is link to the smaller one and put the relevant type name in the displayed text for the link.

**Comment by toteki** [comment]
Heads up analytics (@khoerling) frontend (@EbrahimUmee) - query changes here, most importantly eliminating solo queries for anything that is a field in market summary.

For frontend, note that just like `available_borrow`, there are also `available_withdraw` and `available_collateralize` fields that could place upper limits on whale-sized transactions (or any transactions if market conditions were extreme).

In addition, a lot of the new market summary fields could make good graphs if we're really getting into market details (e.g. graphing `liquidity` vs `collateral * uToken exchange rate` and `minimum_liqudity`.

File: x/leverage/client/cli/query.go [hunk 2]
```diff
@@ -224,33 +216,6 @@ func GetCmdQuerySuppliedValue() *cobra.Command {
 	return cmd
 }
 
-// GetCmdQueryReserveAmount creates a Cobra command to query for the
-// reserved amount of a specific token.
-func GetCmdQueryReserveAmount() *cobra.Command {
```
Comment (RafilxTenfen) [inline x/leverage/client/cli/query.go:229]: I like the idea of condensing queries, but why remove the specific ones? 
Sometimes the user can need only part of that full information '-'

**Review by RafilxTenfen — COMMENTED** [review]
Right now, if this was merged, the user cannot retrieve what we have in the TODO's
neither with the specific queries 

Maybe we could let the specific ones, until we finish the TODO's or even let both

File: x/leverage/client/cli/query.go (hunk 2 above)
Comment (toteki) [reply x/leverage/client/cli/query.go:229]: > but why remove the specific ones?

The idea is to reduce the API surface so there's less to maintain - that said, it's possible that this measure is a bit aggressive. We could keep the specific ones as well as the improved market summary.

File: x/leverage/keeper/grpc_query.go [hunk 3]
```diff
@@ -541,57 +381,48 @@ func (q Querier) MarketSummary(
...
+		Borrowed:               borrowed.Amount,
+		Liquidity:              balance.Sub(reserved),
+		MaximumBorrow:          supplied.Amount, // TODO #1162 #1163 - implement limits
+		MaximumCollateral:      uSupply.Amount,  // TODO #1163 - implement limits
+		MinimumLiquidity:       sdk.ZeroInt(),   // TODO #1163 - implement limits
+		UTokenSupply:           uSupply.Amount,
+		AvailableBorrow:        availableBorrow,
+		AvailableWithdraw:      uSupply.Amount,                  // TODO #1163 - implement limits
+		AvailableCollateralize: uSupply.Amount.Sub(uCollateral), // TODO #1163 - implement limits
+	}
```
Comment (toteki) [inline x/leverage/keeper/grpc_query.go:420]: Note that the current fields in market summary are all working - only the new ones having to do with safety limits have TODOs.

I would prefer to do them in a separate PR, since they'll require some math and module behavior changes. This PR would meanwhile merge everything proto-breaking, reducing the impact of the implementation PR.

File: x/leverage/client/cli/query.go (hunk 2 above)
Comment (RafilxTenfen) [reply x/leverage/client/cli/query.go:229]: It could increase the gas consumption if someone doesn't need all those fields, but I don't know if that is a big problem

Comment (RafilxTenfen) [reply x/leverage/client/cli/query.go:229]: The part of less to maintain is definitely significant on all sides...

**Review by RafilxTenfen — APPROVED** [review]
